restful.module

```diff
@@ -749,3 +768,21 @@ function restful_date_time_format_element_validate($element, &$form_state) {
     )));
   }
 }
+
+/**
+ * Determine if this is the first time we try to switch the user.
+ *
+ * @return bool
+ *   TRUE if this static function was already called.
+ */
+function restful_is_user_switched() {
```

Comment:
What if we made `AuthenticationManager` to be a singleton? https://gist.github.com/mateu-aguilo-bosch/884b3c997f9fb3eef07c

That way we can be **sure** that there is only one `AuthenticationManager`.